File: backend/src/modules/treino_exercicio/entity/TreinoExercicio.js
class TreinoExercicio {
  constructor({ series, repeticoes, treino_id_treino, exercicio_id_exercicio }) {
    this.series = series;
    this.repeticoes = repeticoes;
    this.treino_id_treino = treino_id_treino;
    this.exercicio_id_exercicio = exercicio_id_exercicio;
  }
  validate() {
    const errors = [];
    if (!this.treino_id_treino) errors.push('Treino é obrigatório');
    if (!this.exercicio_id_exercicio) errors.push('Exercício é obrigatório');
    return errors;
  }
}
module.exports = TreinoExercicio;

File: backend/src/main/java/com/vitalativa/entity/TreinoExercicio.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

@Entity
@Table(name = "treino_exercicio")
public class TreinoExercicio {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_treino_exercicio")
    private Integer idTreinoExercicio;
    
    private Integer series;
    private Integer repeticoes;
    
    @NotNull(message = "Treino é obrigatório")
    @Column(name = "treino_id_treino")
    private Integer treinoIdTreino;
    
    @NotNull(message = "Exercício é obrigatório")
    @Column(name = "exercicio_id_exercicio")
    private Integer exercicioIdExercicio;
    
    public TreinoExercicio() {}
    
    public Integer getIdTreinoExercicio() { return idTreinoExercicio; }
    public void setIdTreinoExercicio(Integer idTreinoExercicio) { this.idTreinoExercicio = idTreinoExercicio; }
    public Integer getSeries() { return series; }
    public void setSeries(Integer series) { this.series = series; }
    public Integer getRepeticoes() { return repeticoes; }
    public void setRepeticoes(Integer repeticoes) { this.repeticoes = repeticoes; }
    public Integer getTreinoIdTreino() { return treinoIdTreino; }
    public void setTreinoIdTreino(Integer treinoIdTreino) { this.treinoIdTreino = treinoIdTreino; }
    public Integer getExercicioIdExercicio() { return exercicioIdExercicio; }
    public void setExercicioIdExercicio(Integer exercicioIdExercicio) { this.exercicioIdExercicio = exercicioIdExercicio; }
}